rosdep/python.yaml

```diff
@@ -465,6 +465,10 @@ pyqt5-dev-tools:
   openembedded: ['${PYTHON_PN}-pyqt5@meta-qt5']
   rhel: [python3-qt5-devel]
   ubuntu: [pyqt5-dev-tools]
+pyqt6-dev-tools:
+  debian: [pyqt6-dev-tools]
+  fedora: [python3-qt6-devel]
```

Comment:
So fedora is an interesting situation here.

First of all, the package name would be `python3-pyqt6-devel` (CI will likely fail because of this).

But even if we switch to that, `python3-pyqt6-devel` on Fedora doesn't actually seem to correspond to `pyqt6-dev-tools`  on Ubuntu.  In particular, the "tools" package on Ubuntu seems to provide `/usr/bin/pylupdate6`, `/usr/bin/pyuic6`, as well as `/usr/lib/python3/dist-packages/PyQt6/uic` python code.  The equivalent on Fedora would seem to be `python3-pyqt6-base`.  So I'm going to suggest we switch to that for this key instead.

Reply:
+1 for `python3-pyqt6-base`: https://packages.fedoraproject.org/pkgs/python-pyqt6/python3-pyqt6-base/

Reply:
Will push changes for this tomorrow morning once back at work, only seeing this at EOD.

Reply:
Updated Fedora package name as requested in [c5c5847](https://github.com/ros/rosdistro/pull/40750/commits/c5c5847c402f8720a22abdb2a1c001ce6795b837)